synthesizer/src/process/prepare.rs

```diff
@@ -0,0 +1,148 @@
+
+        // Retrieve the input types.
+        let input_types = self.get_program(program_id)?.get_function(&function_name)?.input_types();
+        // Construct the inputs.
+        let inputs = [Value::Record(credits), Value::from_str(&U64::<N>::new(fee_in_microcredits).to_string())?];
+        lap!(timer, "Construct the inputs");
+        // Compute the request.
+        let request = Request::sign(private_key, program_id, function_name, inputs.iter(), &input_types, rng)?;
+        lap!(timer, "Compute the request");
+        // Initialize the authorization.
+        let authorization = Authorization::new(&[request.clone()]);
+        lap!(timer, "Initialize the authorization");
+        // Construct the call stack.
+        let call_stack = CallStack::Authorize(vec![request], *private_key, authorization.clone());
+        // Construct the authorization from the function.
+        let _response = self.get_stack(program_id)?.execute_function::<A>(call_stack)?;
+        lap!(timer, "Construct the authorization from the function");
+
+        // Retrieve the main request (without popping it).
+        let request = authorization.peek_next()?;
+        // Prepare the stack.
+        let stack = self.get_stack(request.program_id())?;
+
+        #[cfg(feature = "aleo-cli")]
+        println!("{}", format!(" • Calling '{}/{}'...", request.program_id(), request.function_name()).dimmed());
+
+        // Initialize the execution.
+        let execution = Arc::new(RwLock::new(Execution::new()));
+        // Initialize the inclusion.
+        let inclusion = Arc::new(RwLock::new(Inclusion::new()));
+        // Initialize the metrics.
+        let metrics = Arc::new(RwLock::new(Vec::new()));
+        // Initialize the call stack.
+        let assignments = Assignments::<N>::default();
+        // Pass registers to Callstack
+        let call_stack = CallStack::prepare(
+            authorization,
+            execution.clone(),
+            inclusion.clone(),
+            metrics.clone(),
+            assignments.clone(),
+        )?;
+        // Execute the circuit.
+        let response = stack.execute_function::<A>(call_stack)?;
+        lap!(timer, "Execute the circuit");
+
+        // Extract the execution.
+        let execution = Arc::try_unwrap(execution).unwrap().into_inner();
+        // Ensure the execution contains 1 transition.
+        ensure!(execution.len() == 1, "Execution of '{}/{}' does not contain 1 transition", program_id, function_name);
+        // Extract the inclusion.
+        let inclusion = Arc::try_unwrap(inclusion).unwrap().into_inner();
+        // Extract the metrics.
+        let metrics = Arc::try_unwrap(metrics).unwrap().into_inner();
+        // Extract the assignments
+        let assignments = Arc::try_unwrap(assignments).unwrap_or_default().into_inner();
+
+        finish!(timer);
+
+        Ok((response, execution.peek()?.clone(), inclusion, assignments, metrics))
```

Comment:
why not extract (move) the value from `execution` instead of cloning it?

Reply:
Done: cd0a0b10f